src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -39,8 +39,20 @@ public ModelTypeProviderFields(InputModelType inputModel, TypeFactory typeFactor
 
             var visitedMembers = new HashSet<ISymbol>(SymbolEqualityComparer.Default);
 
-            foreach (var inputModelProperty in inputModel.Properties)
+            foreach (var property in inputModel.Properties)
             {
+                var inputModelProperty = property;
+                var name = BuilderHelpers.DisambiguateName(inputModel.Name.ToCleanName(), property.Name);
```

Comment:
here the `DisambiguateName` method append `Value` to the property name when it collides with the type name, which is OK.
If you read the code carefully, you will find we have code from L56-L60, and I quote here:
```csharp
var originalFieldName = inputModelProperty.Name.ToCleanName();
if (originalFieldName == inputModel.Name)
{
    originalFieldName = $"{originalFieldName}Property";
}
```
this is trying to append the word `Property` to this name if it is the same as the model - we need to verify if this is working. And since we choose the word `Value` now, we need to remove this piece of code.

Reply:
This code snippet is added recently after I generate the PR. So I am not aware it. Thanks @ArcturusZhang  to inform this. And I tried, change `originalFieldName` is enough.

Reply:
But we still need to handle other Name ambiguate as DisambiguateName handled, so I remove this logic and use DisambiguateName function.